codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -164,4 +184,51 @@ class FileCodesignVisitor {
       processManager: processManager,
     );
   }
+
+  /// Visit and codesign a binary with / without entitlement.
+  ///
+  /// At this stage, the virtual [entitlementCurrentPath] accumulated through the recursive visit, is compared
+  /// against the paths extracted from [fileWithEntitlements], to help determine if this file should be signed
+  /// with entitlements.
+  Future<void> visitBinaryFile({required File binaryFile, required String entitlementParentPath}) async {
+    if (io.FileSystemEntity.isLinkSync(binaryFile.absolute.path)) {
```

Comment:
I actually don't think this is possible, since we should only get here if `file --mime-type /path/to/entity` returns `application/x-mach-binary`, while a link would return `inode/symlink`. I think you can omit this, or if you want to be safe, instead throw a `StateError('Expected a binary but got a link at ${binaryFile.path}');`

Reply:
Yeah i was confused by this previously as well. For some reasons `mime type` check would not pick up the symlink. it returns a directory type for FlutterMacOS.framework.zip/Versions/Current , which should be
 a link. But for some reasons the symlink check would pick it up and output the correct information.
<img width="903" alt="Screen Shot 2022-08-10 at 4 19 49 PM" src="https://user-images.githubusercontent.com/38773539/184039409-94931c1b-b5d2-4578-88d5-0805ca6deb5c.png">

Reply:
Right, but `FlutterMacOS.framework.zip/Versions/Current` will never his this method, right?

Reply:
True, I will move this check to the directory level

Reply:
Sounds good.

The point is that rather than quietly handling this state, I suspect if we ever hit it there was a bug elsewhere, and it's better to crash than to quietly do the wrong thing.

Reply:
umm previously we implemented a crash on loop (when symlink directs back to parent folder etc.), I thought that covered the crash by symlinks. Umm maybe I am wrong and I am thinking, is there another bug that could be caused by symlink

Reply:
Actually I rethought about it, since symlinks are quite often in engine artifacts, I think Christopher is right, it's a good idea to follow the symlinks, and if they form a cycle, let them crash. early termination on symlink will bypass the loop check which isn't ideal. I have removed this early termination check on symlinks.